carstore/carwriter.go

```diff
@@ -115,7 +115,7 @@ func (cw *CarWriter) write(ctx context.Context, adCid cid.Cid, ad schema.Adverti
 			return nil, err
 		}
 		// OK, car file does not exist.
-	} else {
+	} else if !overWrite {
```

Comment:
Should the flag be the other way around? I.e. if `overwrite` is `true` then remove ad data?

Reply:
No. The call to `removeAdData` removes the ad data from the data store, that was put there by datatransfer/graphsync, instead of writing the car file and then removing that data after writing the car file.

In otherwords, if overWrite is false then only do datastore cleanup without overwriting car file.

Reply:
Added comment to clarify.